### ISSUE TITLE
Rename project/repository

### ISSUE THREAD

**Issue by bjornwgnr** [paritytech/parity-wasm#215]
From a discussion w/ @folsen parity-wasm is neither a random name nor a very descriptive one.

**Comment by pepyakin** [comment]
Indeed. 

As a history note: 
Initially this library had serialization-deserialization capabilities and an interpreter. At the time when we thought how exactly we should proceed with extracting an interpreter from this repo, we evaluated several options, including renaming serialization-deserialization (this repo) to something else. However, we decided to not do it. This is because it seemed that almost nobody used the interpreter, in contrast to serialization-deserialization part of the library. 
Nowadays this still seems to be more or less the case: take a look at [wasmi](https://crates.io/crates/wasmi/reverse_dependencies) and [parity-wasm](https://crates.io/crates/parity-wasm/reverse_dependencies) reverse dependencies (However, I must admit that it is possible that `parity-wasm` is more suitable for libraries than for end applications, unlike `wasmi`).

Basically, this is the reason we created new repo for the [wasmi](https://github.com/paritytech/wasmi) and just removed the interpreter and left the name as is for [parity-wasm](https://github.com/paritytech/parity-wasm).

**Comment by NikVolf** [comment]
We shouldn't rename this library, name is more or less decent and it's long on crates.io already

**Comment by folsen** [comment]
We need to consider an overall naming scheme, there's like 6 different repos with different naming structures and it's impossible to figure out what is what. Is `parity-wasm` the same as `pWasm`? What is then `pWasm`? Is `parity-wasm` a library from parity for general wasm stuff, for eWasm stuff?

It's impossible to tell what `parity-wasm` is or what it's supposed to be, but it's not just this repo in isolation that is the issue, the naming of all things in general need to be looked over. Like if it's generally useful serialisation and deserialisation of wasm, then why is it called `parity-`, people will avoid using it because they think it's only useful for parity.

I don't have a great suggestion, but I know that there's tons of people out there that are confused.

**Comment by NikVolf** [comment]
Nobody from those who really need format library is confused, they just use it

Parity means something mostly for us, for other users it's pretty random name already

**Comment by folsen** [comment]
The problem is more in the other direction, people coming from cryptocurrency side trying to find our wasm interpreter or Rust smart contract libraries and they have no way of finding it and this is what they get.
